### PR TITLE
Makefile,hack/tests: separate tests to make testing simpler and quicker 

### PR DESCRIPTION
**Pull request by AlexNPavel**
This separates the tests into multiple different scripts based on their purpose. This allows developers to more quickly and easily test their work. It also makes it easier to run the tests in parallel, which can help speed up CI.

### CODE REVIEW

File: hack/tests/e2e-ansible.sh [hunk 1]
```diff
@@ -53,9 +93,5 @@ then
     exit 1
 fi
 
-# Test finalizer
-kubectl delete -f deploy/crds/ansible_v1alpha1_memcached_cr.yaml --wait=true
```
[the remark this answers is not in the document]
Comment (joelanford) [reply hack/tests/e2e-ansible.sh]: Just tested this by adding `; false` to the end of the trapped statement and the script does exit with a failed error code.

It looks like the trap commands run in reverse order based on when they're added and any failure will result in the remaining commands not being run. I would think we'd leave the tests out of traps with the reasoning that if a test fails, we still want to clean up after ourselves.

Maybe the best solution here is for the trap command to be:
```bash
trap_add 'kubectl delete --ignore-not-found -f ${DIR2}/deploy/crds/ansible_v1alpha1_memcached_cr.yaml' EXIT
```
And then we leave the finalizer test commands in place.

**Comment by mhrivnak** [comment]
More curious than anything else: why move the tests into a directory called `hack`?

**Comment by AlexNPavel** [comment]
> More curious than anything else: why move the tests into a directory called `hack`?

That's just where we put most of our testing scripts. I've seen various go repositories use a `hack` folder for shell scripts, and we already had one so it seemed to make sense placing the script in there.

**Pull request merged**